pkgs/build-support/dotnet/add-packagelock-deps/default.nix

```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  fetchNupkg,
+  runCommand,
+  callPackage,
+}:
+
+{
+  nugetLockFile,
+  overrideFetchAttrs ? x: { },
+}:
+fnOrAttrs: finalAttrs:
+let
+  attrs = if builtins.isFunction fnOrAttrs then fnOrAttrs finalAttrs else fnOrAttrs;
+  dependencyTool = callPackage ./tool.nix { };
+
+  deps =
+    with builtins;
+    let
+      projectFiles = attrs.dotnetProjectFiles ++ attrs.dotnetTestProjectFiles;
+      dynamicLockFiles = fromJSON (
+        readFile (
+          runCommand "lockfile-discovery"
+            {
+              inherit projectFiles;
+              sourceDir = "${attrs.src}";
```

Comment:
I should probably add something like `Configuration = attrs.dotnetBuildType` here. Maybe wrapping in `lib.optionalAttrs` to allow the project file to use its own default when it's not specified in the nix config.